Extensions/TweenBehavior/tweenruntimebehavior.ts

```diff
@@ -58,119 +100,102 @@ namespace gdjs {
       this._runtimeScene = runtimeScene;
     }
 
-    updateFromBehaviorData(oldBehaviorData, newBehaviorData): boolean {
+    updateFromBehaviorData(
+      oldBehaviorData: BehaviorData,
+      newBehaviorData: BehaviorData
+    ): boolean {
       // Nothing to update.
       return true;
     }
 
-    _addTween(identifier, instance, startTime, totalDuration) {
+    private _addTween(
+      identifier: string,
+      easingValue: string,
+      tweenConfig: shifty.tweenConfig,
+      startTime: number,
+      totalDuration: number,
+      destroyObjectWhenFinished: boolean
+    ): void {
+      if (!this._isActive || !!TweenRuntimeBehavior.easings[easingValue])
+        return;
+
+      // Remove any prior tween
+      if (this._tweenExists(identifier)) this.removeTween(identifier);
+
+      // Initialize the tween instance
+      const tweenable = new shifty.Tweenable();
       this._tweens[identifier] = new TweenRuntimeBehavior.TweenInstance(
-        instance,
+        tweenable,
         false,
         startTime,
         totalDuration
       );
-    }
 
-    _getTween(identifier) {
-      return this._tweens[identifier];
+      // Attach it to the scene as it will become active and acttive tweens should be on the scene.
+      if (!this._runtimeScene.shiftyJsScene)
+        this._runtimeScene.shiftyJsScene = new shifty.Scene();
+      this._runtimeScene.shiftyJsScene.add(tweenable);
+
+      // Start the tween and set the needed callbacks
+      tweenable
+        .tween(tweenConfig)
+        .then(() => {
+          if (this._tweens[identifier])
+            this._tweens[identifier].hasFinished = true;
+
+          if (destroyObjectWhenFinished)
+            this.owner.deleteFromScene(this._runtimeScene);
+        })
+        .catch((e) => {
```

Comment:
Shifty was at least in the past throwing errors when you attempted to stop a tween and restart it if it was not finished (despite not being an "error"/"exception" by itself - it's a valid use case).
Is this still the case? If so we should not log anything.

Reply:
@arthuro555 We'll need to be sure about this before going ahead with the PR :)

Reply:
Yes, to need to test this

Reply:
Neither that: 
![image](https://user-images.githubusercontent.com/19349038/141696474-bb556593-ad96-4b92-adb5-4174b7d6c003.png)
Nor that (whether the tween was playing before or not):
![image](https://user-images.githubusercontent.com/19349038/141696494-1fd12f02-7cf3-4876-9d04-459de6f51717.png)

Logged any error in the console, I am assuming that this behavior has since changed in shifty.

Reply:
@jeremyckahn hope it's ok to quickly ping you to have your blessing on this: we're making the assumption that tweens won't "throw" in Shifty - is this ok to do?

I see in Shifty code that the promise can be rejected, but only in the case where we actually defined a catch/reject handler:
![image](https://user-images.githubusercontent.com/1280130/141773005-eb69208e-4bd1-45f8-a3d2-7d6715032a56.png)

Are there other cases where this can happen?
- If yes, then let's probably not log anything (we don't want to spam logs in a game, decreasing performance) if this can happen as part of the normal operation of a game.
- If not, then let's entirely remove the catch.

Reply:
Hi @4ian, thanks for the ping! It's no trouble at all.

The only scenario where I expect a Shifty tween to throw an error is when its `.then` function is called (which will happen implicitly if it is being `await`ed) and it is also `cancel`led:

https://github.com/jeremyckahn/shifty/blob/199774dc8e03b7de1a1971997193e692a5626fe1/src/tweenable.js#L452-L459

https://github.com/jeremyckahn/shifty/blob/199774dc8e03b7de1a1971997193e692a5626fe1/src/tweenable.js#L635-L654

I haven't thoroughly reviewed this PR, but it does seem that there is at least one place where GDevelop is calling a `Tweenable`'s `.then` function: https://github.com/4ian/GDevelop/pull/3218/files#diff-90447101490944a2b12792b193b1f177c27b721e583d175e6eec438a89f232f9R142-R149

FWIW, I believe using Shifty's Promise API is good practice, so that's fine to leave in. However, because of that I think GDevelop should also anticipate the rejection caused by a potential future `cancel` call. It seems to me that the `catch` we're talking about should remain in case a `cancel` call is ever added.

I hope this is helpful. I'm excited to see GDevelop use the latest version of Shifty! Let me know if I can answer any other questions. 🙂

Reply:
> However, because of that I think GDevelop should also anticipate the rejection caused by a potential future cancel call. It seems to me that the catch we're talking about should remain in case a cancel call is ever added.

Sounds good!

@arthuro555 Can we remove the log in the catch, because it could in the future be called for a cancelled tween, which is not an "error" by itself? Then we're good to go!